pageserver/src/tenant/timeline/test_utils.rs

```diff
@@ -64,11 +63,12 @@ impl Timeline {
         let min_key = *deltas.first().map(|(k, _, _)| k).unwrap();
         let max_key = deltas.last().map(|(k, _, _)| k).unwrap().next();
         let min_lsn = *deltas.iter().map(|(_, lsn, _)| lsn).min().unwrap();
-        let max_lsn = Lsn(deltas.iter().map(|(_, lsn, _)| lsn).max().unwrap().0 + 1);
+        let max_lsn = *deltas.iter().map(|(_, lsn, _)| lsn).max().unwrap();
         assert!(
             max_lsn <= last_record_lsn,
             "advance last record lsn before inserting a layer, max_lsn={max_lsn}, last_record_lsn={last_record_lsn}"
         );
+        let max_lsn = Lsn(max_lsn.0 + 1);
```

Comment:
I hate that we do this off by one without using a Range type.

Also I just realized key range is `start..=end` and lsn range is `start..end`.

As for a concrete change, I wonder if we more commonly use start and end. At least here, `end_lsn` could be used as it is off by one, and the max lsn stays the same.

I know I am splitting hairs here, but a knee-jerk reaction from someone who does not know the "convention of off by one" is to say this piece of code is wrong; max_lsn didn't just magically advance by one, yet we did advance it for no obvious reason.

So perhaps:

```patch
-let max_lsn = Lsn(max_lsn.0 + 1);
+let end_lsn = Lsn(max_lsn.0 + 1);
```